test/charts/helm_broker_test.go

```diff
@@ -0,0 +1,188 @@
+
+	cli, err := versioned.NewForConfig(k8sCfg)
+	require.NoError(t, err)
+
+	randID := rand.String(5)
+
+	return &TestSuite{
+		t:   t,
+		cfg: cfg,
+
+		k8sClientCfg: k8sCfg,
+		addonsCli:    cli,
+
+		sampleClusterAddonsCfgName: fmt.Sprintf("test-hb-chart-%s", randID),
+	}
+}
+
+func (s *TestSuite) createSampleClusterAddonsConfiguration() {
+	_, err := s.addonsCli.AddonsV1alpha1().ClusterAddonsConfigurations().Create(&v1alpha1.ClusterAddonsConfiguration{
+		ObjectMeta: v1.ObjectMeta{
+			Name: s.sampleClusterAddonsCfgName,
+		},
+		Spec: v1alpha1.ClusterAddonsConfigurationSpec{
+			CommonAddonsConfigurationSpec: v1alpha1.CommonAddonsConfigurationSpec{
+				Repositories: []v1alpha1.SpecRepository{
+					{URL: s.cfg.ClusterAddonsConfigurationURL},
+				},
+			},
+		},
+	})
+
+	require.NoError(s.t, err, "while creating cluster addons configuration")
+	s.t.Logf("ClusterAddonsConfigurations %q is created", s.sampleClusterAddonsCfgName)
+}
+
+func (s *TestSuite) deleteSampleClusterAddonsConfiguration() {
+	err := s.addonsCli.AddonsV1alpha1().ClusterAddonsConfigurations().Delete(s.sampleClusterAddonsCfgName, &v1.DeleteOptions{})
+	require.NoError(s.t, err, "while creating cluster addons configuration")
+	s.t.Logf("ClusterAddonsConfigurations %q is deleted", s.sampleClusterAddonsCfgName)
+}
+
+func (s *TestSuite) waitForSampleClusterAddonsConfiguration(timeout time.Duration) {
+	sampleClusterAddonsAvailable := func() error {
+		cac, err := s.addonsCli.AddonsV1alpha1().ClusterAddonsConfigurations().Get(s.sampleClusterAddonsCfgName, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if cac.Status.Phase != v1alpha1.AddonsConfigurationReady {
+			return fmt.Errorf("ClusterAddonsConfiguration %q is not in %s state. Current status: %v",
+				s.sampleClusterAddonsCfgName,
+				v1alpha1.AddonsConfigurationReady,
+				cac.Status,
+			)
+		}
+
+		return nil
+	}
+
+	waitForFuncAtMost(s.t, sampleClusterAddonsAvailable, timeout)
+	s.t.Logf("ClusterAddonsConfigurations %q has %s state", s.sampleClusterAddonsCfgName, v1alpha1.AddonsConfigurationReady)
+}
+
+func (s *TestSuite) assertSampleClusterAddonsAreExposedOnCatalogEndpoint(timeout time.Duration) {
+	config := osb.DefaultClientConfiguration()
+	config.URL = s.cfg.HelmBrokerURL
+
+	client, err := osb.NewClient(config)
+	require.NoError(s.t, err, "while creating osb client for broker with URL: %s", s.cfg.HelmBrokerURL)
+
+	response, err := client.GetCatalog()
+	require.NoError(s.t, err, "while getting catalog from broker with URL: %s", s.cfg.HelmBrokerURL)
+
+	containService := func(id string) bool {
+		for _, svc := range response.Services {
+			if svc.ID == id {
+				return true
+			}
+		}
+		return false
+	}
+
+	assert.True(s.t, containService(s.cfg.ExpectedAddonID))
+	s.t.Logf("Helm Broker exposes Service [id: %q] from ClusterAddonsConfiguration %s properly", s.cfg.ExpectedAddonID, s.sampleClusterAddonsCfgName)
+}
+
+func waitForFuncAtMost(t *testing.T, fn func() error, timeoutDur time.Duration) {
```

Comment:
We use `k8s.io/apimachinery/pkg/util/wait` library in an integration test, is it any reason why you use a custom function here? You don't like apimachinery function ;) ?